tests/test_runner.py

```diff
@@ -587,3 +587,16 @@ def test_invalid_sleep():
                 with pytest.raises(type(e)):
                     runner.sleep(arg)
                 raise
+
+def test_snoop():
+    code = dedent(
+        """
+    x = 5
+    x *= 2
+        """)
+    check_simple(code, [
+        ('output', {'parts': [{'text': '','type': 'snoop'}]}),
+        ('output', {'parts': [{'text': '    2 | x = 5\n', 'type': 'snoop'}]}),
+        ('output', {'parts': [{'text': '    3 | x *= 2\n', 'type': 'snoop'}]}),
+        ('output', {'parts': [{'text': ' ...... x = 10\n', 'type': 'snoop'}]})
```

Comment:
Oooooh, I'm glad we have this test because this reveals a mild bug I didn't know about. `OutputBuffer` should be combining these parts, but snoop calls `flush` after each write. I think `SysStream.flush` should do nothing for the `snoop` type.

Anyway, it would still be good for the code to have a (trivial) function in it to test the recursive tracing.